### PR TITLE
Protect the stage2 device from the Storage module

### PR DESCRIPTION
**Pull request by poncovka**
Move the code for protection of the stage2 device to the Storage module.
The device will be detected and protected on every storage reset.

**Removal of the SourceFactory class - part 2**

### CODE REVIEW

**Review by M4rtinK — APPROVED** [review]
Looks good to me. :)

**Review by VladimirSlavik — APPROVED** [review]
Looks good to me. Thank you!

File: pyanaconda/modules/storage/devicetree/model.py [hunk 1]
```diff
@@ -289,6 +289,13 @@ def _mark_protected_devices(self):
                 log.debug("Protected device spec %s resolved to %s.", spec, dev.name)
                 protected.append(dev)
 
+        # Find the stage2 backing device and its parents.
+        stage2_device = find_stage2_device(self.devicetree)
+
+        if stage2_device:
+            log.debug("Resolved stage2 device to %s.", stage2_device.name)
+            protected.append(stage2_device)
```
[the remark this answers is not in the document]
Comment (VladimirSlavik) [reply pyanaconda/modules/storage/devicetree/model.py:297]: Oops, sorry, missed that bit.

**Pull request merged**